### PR TITLE
cp-25:  Mobile Protected Routing

### PR DESCRIPTION
**Pull request by danilittaa**
- created ProtectedRoute as described in issue

### CODE REVIEW

File: mobile/src/libs/components/protected-route/protected-route.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+const ProtectedRoute: React.FC<Proptertis> = ({ children }) => {
+  const { user } = useAppSelector((store) => store.auth);
+  const navigation =
+    useNavigation<NativeStackNavigationProp<RootNavigationParameterList>>();
+  const hasUser = Boolean(user?.id);
+
+  useEffect(() => {
+    if (!hasUser) {
+      navigation.navigate('Sign In');
```
Comment (v0ldemar01) [inline mobile/src/libs/components/protected-route/protected-route.tsx]: Please add an enum value

**Review by v0ldemar01 — CHANGES_REQUESTED** [review]

File: mobile/src/libs/components/app/app.tsx [hunk 2]
```diff
@@ -21,7 +22,9 @@ const App: React.FC = () => {
     <StoreProvider store={store.instance}>
       <GestureHandlerRootView style={styles.root}>
         <NavigationContainer>
-          <RootNavigation />
+          <ProtectedRoute>
+            <RootNavigation />
+          </ProtectedRoute>
```
Comment (fshabanov) [inline mobile/src/libs/components/app/app.tsx]: We don't want to make every screen protected. Screens like Sign in and Sign up should not be protected.

File: mobile/src/libs/components/protected-route/protected-route.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { type NativeStackNavigationProp } from '@react-navigation/native-stack/lib/typescript/src/types';
+import React, { type ReactNode } from 'react';
+
+import { useAppSelector, useEffect, useNavigation } from '#libs/hooks/hooks';
+import { type RootNavigationParameterList } from '#libs/types/types';
+
+type Proptertis = {
```
Comment (fshabanov) [inline mobile/src/libs/components/protected-route/protected-route.tsx]:
```suggestion
type Properties = {
```

File: mobile/src/libs/components/protected-route/protected-route.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  children: ReactNode;
+};
+
+const ProtectedRoute: React.FC<Proptertis> = ({ children }) => {
+  const { user } = useAppSelector((store) => store.auth);
+  const navigation =
+    useNavigation<NativeStackNavigationProp<RootNavigationParameterList>>();
+  const hasUser = Boolean(user?.id);
```
Comment (fshabanov) [inline mobile/src/libs/components/protected-route/protected-route.tsx]:
```suggestion
  const hasUser = Boolean(user);
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: mobile/src/libs/components/protected-route/protected-route.tsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+import { RootScreenName } from '#libs/enums/enums';
+import { useAppSelector, useEffect, useNavigation } from '#libs/hooks/hooks';
+import { type RootNavigationParameterList } from '#libs/types/types';
+
+type Properties = {
+  children: ReactNode;
+};
+
+const ProtectedRoute: React.FC<Properties> = ({ children }) => {
+  const { user } = useAppSelector((store) => store.auth);
```
Comment (liza-veis) [inline mobile/src/libs/components/protected-route/protected-route.tsx]:
```suggestion
  const { authenticatedUser } = useAppSelector(({auth}) => ({
    authenticatedUser: auth.authenticatedUser
  }));
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Review by OlenaYatseniuk — DISMISSED** [review]

**Review by liza-veis — APPROVED** [review]

File: mobile/src/navigations/root/root.tsx [hunk 6]
```diff
@@ -21,7 +22,14 @@ const Root: React.FC = () => {
     <NativeStack.Navigator screenOptions={screenOptions}>
       <NativeStack.Screen name={RootScreenName.SIGN_IN} component={Auth} />
       <NativeStack.Screen name={RootScreenName.SIGN_UP} component={Auth} />
-      <NativeStack.Screen name={RootScreenName.MAIN} component={Main} />
+      <NativeStack.Screen
+        name={RootScreenName.MAIN}
+        component={(): JSX.Element => (
+          <ProtectedRoute>
```
Comment (liza-veis) [inline mobile/src/navigations/root/root.tsx]: Let's leave this component but don't wrap anything for now, we will use it a bit later

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**